src/megaapi_impl.cpp

```diff
@@ -14216,73 +14278,155 @@ void MegaApiImpl::getua_result(byte* data, unsigned len, attr_t type)
+                            Base64::atob(sValue, bValue);
+                            request->setFlag(bValue == "1");
+                        }
+                        else
+                        {
+                            request->setFlag(false);
+                        }
                     }
+                    else if (request->getNumDetails() == 1) // used to check if should show warning
+                    {
+                        request->setFlag(!num);
+                        // it doesn't matter the value, just if it exists
+
+                        const char *value = stringMap->get("c");
+                        if (value)
+                        {
+                            string sValue = value;
+                            string bValue;
+                            Base64::atob(sValue, bValue);
+                            request->setNumber(atoi(bValue.c_str()));
+                        }
+                    }
+                    break;
                 }
-                else if (request->getNumDetails() == 1) // used to check if should show warning
+                case MegaApi::USER_ATTR_CAMERA_UPLOADS_FOLDER:
+                case MegaApi::USER_ATTR_MY_CHAT_FILES_FOLDER:
                 {
-                    request->setFlag(!num);
-                    // it doesn't matter the value, just if it exists
-
-                    const char *value = stringMap->get("c");
+                    const char *value = stringMap->get("h");
                     if (value)
                     {
-                        string sValue = value;
-                        string bValue;
-                        Base64::atob(sValue, bValue);
-                        request->setNumber(atoi(bValue.c_str()));
+                        handle nodehandle;
+                        Base64::atob(value, (byte*) &nodehandle, MegaClient::NODEHANDLE);
+                        request->setNodeHandle(nodehandle);
                     }
+                    break;
                 }
-                break;
-            }
-            case MegaApi::USER_ATTR_CAMERA_UPLOADS_FOLDER:
-            case MegaApi::USER_ATTR_MY_CHAT_FILES_FOLDER:
-            {
-                const char *value = stringMap->get("h");
-                if (value)
+
+                case MegaApi::USER_ATTR_ALIAS:
                 {
-                    handle nodehandle;
-                    Base64::atob(value, (byte*) &nodehandle, MegaClient::NODEHANDLE);
-                    request->setNodeHandle(nodehandle);
+                    // If a uh was set in the request we have to find it in the aliases map and return it
+                    const char *uh = request->getText();
+                    if (uh)
+                    {
+                        const char *buf, *key;
+                        bool found = false;
+
+                        MegaStringList *keys = stringMap->getKeys();
+                        for (int i=0; i < keys->size(); i++)
```

Comment:
Why do we iterate here, instead of simply get the value for the known key in the map?
```request->setName(stringMap->get(uh));```